java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java

```diff
@@ -42,15 +43,23 @@
   private static final Logger LOG = Loggers.get(SymbolicExecutionVisitor.class);
 
   @VisibleForTesting
-  public final BehaviorCache behaviorCache;
+  public BehaviorCache behaviorCache;
   private final ExplodedGraphWalker.ExplodedGraphWalkerFactory egwFactory;
+  private SquidClassLoader classLoader;
 
   public SymbolicExecutionVisitor(List<JavaFileScanner> executableScanners) {
     this(executableScanners, new SquidClassLoader(Lists.newArrayList()));
   }
+
   public SymbolicExecutionVisitor(List<JavaFileScanner> executableScanners, SquidClassLoader classLoader) {
-    behaviorCache = new BehaviorCache(this, classLoader);
     egwFactory = new ExplodedGraphWalker.ExplodedGraphWalkerFactory(executableScanners);
+    this.classLoader = classLoader;
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    behaviorCache = new BehaviorCache(this, classLoader, (SemanticModel) context.getSemanticModel());
```

Comment:
getSemanticModel might return a null value (both corner case in case of semantic analysis failure (can happen) or when analyzing some jdk classes (less likely)).